### PR TITLE
ERC20 metadata extension

### PR DESCRIPTION
**Pull request by tomasienrbc**
This is the initial submission for an ERC20 Metadata Extension standard. The goal of this EIP is to optionally extend ERC20 token interface to support the same metadata standard as ERC721 tokens. This supports fungible crypto-collectibles that have the same metadata structure as non-fungible crypto-collectibles, amongst many other use cases. 

This was a collaboration between @shrugs, @JohnZettler, and @mateosu

### CODE REVIEW

File: EIPS/eip-721.md [hunk 1]
```diff
@@ -186,13 +186,13 @@ interface ERC721Metadata /* is ERC721 */ {
 
     /// @notice A distinct Uniform Resource Identifier (URI) for a given asset.
     /// @dev Throws if `_tokenId` is not a valid NFT. URIs are defined in RFC
-    ///  3986. The URI may point to a JSON file that conforms to the "ERC721
```
Comment (Arachnid) [inline EIPS/eip-721.md]: Please don't include changes to unrelated files in your PR.

File: EIPS/eip-erc20-metadata-extension.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+---
+eip: <to be assigned>
```
Comment (Arachnid) [inline EIPS/eip-erc20-metadata-extension.md]: Please number this 1046 and rename eip-1046.md.

File: EIPS/eip-erc20-metadata-extension.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+title: ERC20 Metadata Extension
+author: Tommy Nicholas (@tomasienrbc), Matt Russo (@mateosu), John Zettler (@JohnZettler), Matt Condon (@shrugs)
+discussions-to: thelab@rareart.io
```
Comment (Arachnid) [inline EIPS/eip-erc20-metadata-extension.md]: Please supply a mailing list or discussion forum URL.

File: EIPS/eip-erc20-metadata-extension.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+status: Draft
+type: Standards Track
+category: ERC
+created: 2018-04-13
+requires: 20
+---
+
+## Simple Summary
+Optionally extend ERC20 token interface to support the same metadata standard as ERC721 tokens
```
Comment (Shrugs) [inline EIPS/eip-erc20-metadata-extension.md]: nit: period

File: EIPS/eip-erc20-metadata-extension.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+## Motivation
+The ERC721 standard was created to support the creation of perfectly unique, 1-of-1, non-divisible tokens known as "non-fungible tokens".
+
+The initial use case for the ERC721 standard was to support the creation of crypto-collectibles and gaming assets, initially for the ["Crypto Kitties"](https://www.cryptokitties.co/) collectibles game. The success of Crypto Kitties catalyzed significant application development to support the display of ERC721 assets using the `tokenURI` metadata parameter.
+
+However, not all crypto-collectibles and gaming assets need to be unique and non-fungible. Gaming assets (items, weapons, characters), crypto-artworks with non-unique "prints", and more will function more like traditional ERC20 tokens with a fungible `supply`. Many applications such as wallets, exchanges, games, etc. will want to support both fungible and non-fungible assets containing similar metadata. This proposal will extend the ERC20 standard to optionally include a nearly identical `tokenURI` parameter supporting the same JSON metadata schema as the ERC721 standard.
```
Comment (Shrugs) [inline EIPS/eip-erc20-metadata-extension.md]: why "nearly" and not just identical?

File: EIPS/eip-erc20-metadata-extension.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            "type": "string",
+            "description": "A URI pointing to a resource with mime type image/* representing the asset to which this token represents. Consider making any images at a width between 320 and 1080 pixels and aspect ratio between 1.91:1 and 4:5 inclusive.",
+        }
+    }
+}
+```
+## Rationale
+This proposal will make adding metadata to ERC20 tokens straightforward for developers with minimal-to-no disruption to the overall ecosystem. By using the same parameter name and by consolidating the underlying Token JSON Metadata Standard, developers will confidently understand how to add and interpret token metadata between ERC20 and ERC721 tokens.
+
+## Backwards Compatibility
```
Comment (Shrugs) [inline EIPS/eip-erc20-metadata-extension.md]: Mention that the token's `name()` and `symbol()` getters should be preferred over the metadata's `name` and (optional) `symbol` properties?

Comment (tomasienrbc) [reply EIPS/eip-erc20-metadata-extension.md]: 👍

File: EIPS/eip-1046.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
+---
+eip: 1046
+title: ERC20 Metadata Extension
+author: Tommy Nicholas (@tomasienrbc), Matt Russo (@mateosu), John Zettler (@JohnZettler), Matt Condon (@shrugs)
+discussions-to: thelab@rareart.io
```
Comment (Arachnid) [inline EIPS/eip-1046.md]: Please remove this, and rename `discussion-forum` to `discussions-to`.

**Comment by eip-automerger** [comment]
Hi! I'm a bot, and I wanted to automerge your PR, but couldn't because of the following issue(s):

 - Contains new file EIPS/eip-1046.md
 - EIP 721 requires approval from one of (nastassia.sachs@protonmail.com, @fulldecent, @dete, @dekz)

**Comment by tomasienrbc** [comment]
@Arachnid I accidentally have a file from ERC721 changed in this PR (sorry didn't catch that) - removing that and re-submitting

**Comment by tomasienrbc** [comment]
@Arachnid changes to ERC721 are now fully reverted - no changes to ERC721 in this PR, auto-merge should be possible

File: EIPS/eip-1046.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
+---
+eip: 1046
+title: ERC20 Metadata Extension
+author: Tommy Nicholas (@tomasienrbc), Matt Russo (@mateosu), John Zettler (@JohnZettler), Matt Condon (@shrugs)
+discussion-to: https://www.reddit.com/r/raredigitalart/comments/8hfh1g/erc20_metadata_extension_eip_1046/
```
Comment (Arachnid) [inline EIPS/eip-1046.md]: This should be `discussions-to`.

Comment (tomasienrbc) [reply EIPS/eip-1046.md]: Ah yes - fixed
